python/tchannel/handler.py

```diff
@@ -80,9 +80,14 @@ class TChannelRequest(object):
 
     def __init__(self, context, conn):
         self.message = context.message
-        self.header = getattr(self.message, 'arg_2', None)
-        self.body = getattr(self.message, 'arg_3', None)
-        self.method = getattr(self.message, 'arg_1', None)
+
+        try:
+            self.method = self.message.args[0]
+            self.header = self.message.args[1]
+            self.body = self.message.args[2]
+        except:
+            pass
```

Comment:
My concern is the catch-all exception. Check the length and assign attributes instead of a catch-all argument.

Also, you want to make sure these attributes exist regardless of whether the corresponding arg was set. So, you need to give them a default value of None or empty and then assign to them based on how many elements `message.args` contains.